src/plugins/socketmon/socketmon.cpp

```diff
@@ -1443,54 +1434,72 @@ static event_response_t trap_DnsQuery_W_cb(drakvuf_t drakvuf, drakvuf_trap_info_
 {
     socketmon* sm = (socketmon*)info->trap->data;
 
-    addr_t domain_name_addr = 0;
+    addr_t domain_name_addr = drakvuf_get_function_argument(drakvuf, info, 1);
     unicode_string_t* domain_name_us = nullptr;
 
-    if (sm->pm == VMI_PM_IA32E) // only 64 bit for now
-    {
-        domain_name_addr = info->regs->rcx;
-
-        access_context_t ctx;
-        ctx.translate_mechanism = VMI_TM_PROCESS_DTB;
-        ctx.dtb = info->regs->cr3;
-        ctx.addr = domain_name_addr;
+    access_context_t ctx;
+    ctx.translate_mechanism = VMI_TM_PROCESS_DTB;
+    ctx.dtb = info->regs->cr3;
+    ctx.addr = domain_name_addr;
 
-        {
-            vmi_lock_guard vmi_lg(drakvuf);
-            domain_name_us = drakvuf_read_wchar_string(vmi_lg.vmi, &ctx);
-        }
+    {
+        vmi_lock_guard vmi_lg(drakvuf);
+        domain_name_us = drakvuf_read_wchar_string(vmi_lg.vmi, &ctx);
+    }
 
-        if (domain_name_us)
-        {
-            print_dns_info(drakvuf, info, sm, info->trap->name, (char*)domain_name_us->contents);
-        }
-        else
-        {
-            PRINT_DEBUG("[SOCKETMON] Error, getting unicode domain name string in %s()", __FUNCTION__);
-        }
-        vmi_free_unicode_str(domain_name_us);
+    if (domain_name_us)
+    {
+        print_dns_info(drakvuf, info, sm, info->trap->name, (char*)domain_name_us->contents);
     }
     else
     {
-        PRINT_DEBUG("[SOCKETMON] Suddenly, we are in 32 bit mode in %s(...) trap. Unsupported.\n", info->trap->name);
+        PRINT_DEBUG("[SOCKETMON] Error, getting unicode domain name string in %s()", __FUNCTION__);
     }
-
+    vmi_free_unicode_str(domain_name_us);
 
     return 0;
 }
 
+// This is yet another type of Windows string representation
+// specific for undocumented DnsQueryExW(...) function.
+// Same type for 64 and 32 bit versions.
+struct dns_query_ex_w_string_t
+{
+    uint32_t length = 0;
+    uint32_t unknown = 0; // maybe type of bytes in string, was equal to 1 in my case of wchars?
+    uint64_t pBuffer = 0; // pointer to a null-terminated string of wchars
+    //uint64_t unknown2 = 0; // maybe type of bytes in string, was equal to 1 in my case of wchars, commented out, since not needed yet
+};
+
 // Works on Windows 7
 static event_response_t trap_DnsQueryExW_cb(drakvuf_t drakvuf, drakvuf_trap_info_t* info)
 {
+    unicode_string_t* domain_name_us = nullptr;
     socketmon* sm = (socketmon*)info->trap->data;
 
-    if (sm->pm != VMI_PM_IA32E) // only 64 bit for now
+    addr_t domain_name_addr = drakvuf_get_function_argument(drakvuf, info, 1);
+
     {
-        PRINT_DEBUG("[SOCKETMON] Suddenly, we are in 32 bit mode in %s(...) trap. Unsupported.\n", info->trap->name);
-        return 0;
-    }
+        dns_query_ex_w_string_t strange_string;
```

Comment:
This variable name begs for more information :)

Reply:
Renamed into 'function_specific_string'